react-native/update-font-icon.js

```diff
@@ -0,0 +1,44 @@
+// Generate our react-native icons from the fonticon stylesheet and images
+import fs from 'fs'
+import path from 'path'
+
+const css = fs.readFileSync('../desktop/renderer/fonticon.css', {encoding: 'utf8'})
+const stream = fs.createWriteStream('../shared/common-adapters/icon.paths.native.js')
+
+stream.once('open', () => {
+  stream.write('// This file is generated by update-font-icon.js, DO NOT HAND EDIT!\n')
+  stream.write('// @flow\n')
+  stream.write('\n')
+
+  stream.write('export const fontIcons = {\n')
+
+  const glyphs = []
+  let glyph = null
+  css.split('\n').forEach(line => {
+    if (glyph) {
+      const val = line.match(/content:\ \"\\(.*)\";/)[1]
+      glyphs.push(`  '${glyph}': String.fromCharCode(0x${val})`)
+      glyph = null
+    } else {
+      glyph = line.match(/^\.(.*):before/)
+      if (glyph) {
+        glyph = glyph[1]
+      }
+    }
```

Comment:
its just keeping track of the 'next' thing to consume. it looks for a line like
```
.a-name:before {
 the-content-i-care-about
}
```
since this is consuming this a line at a time it bookkeeps, 'i found 'a-name', then next line will have the value for 'a-name'. that's all